src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -197,48 +197,35 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
 
     void ExtensionPalette::_splitResponseAndAddToChatHelper(const IResponse response)
     {
-        // this function is dependent on the AI response separating code blocks with
-        // newlines and "```". OpenAI seems to naturally conform to this, though
-        // we could probably engineer the prompt to specify this if we need to.
-        std::wstringstream ss(response.Message().c_str());
-        std::wstring line;
-        std::wstring codeBlock;
-        bool inCodeBlock = false;
         const auto time = _getCurrentLocalTimeHelper();
         std::vector<IInspectable> messageParts;
 
-        while (std::getline(ss, line))
-        {
-            if (!line.empty())
+        const auto chatMsg = winrt::make<ChatMessage>(winrt::to_hstring(response.Message()), false);
+        chatMsg.RunCommandClicked([this](auto&&, const auto commandlines) {
+            auto suggestion = winrt::to_string(commandlines);
+            // the AI sometimes sends multiline code blocks
+            // we don't want to run any of those commands when the chat item is clicked,
+            // so we replace newlines with the appropriate delimiter
```

Comment:
BTW on that topic again, unrelated to this PR, and related to #18186: I still think we should better integrate this with ControlCore's other paste facilities. It knows whether the shell has enabled bracketed paste (usually true for UNIX shells) which allows for perfect multi-line pastes. For PowerShell without bracketed paste we can still use `\n` as multi-line delimiters. We can infer PowerShell from the profile source (we make similar similar assumptions that `profile_source == WSL` means it wants WSL paths).

Reply:
Agreed, I'll make a follow-up for the way we input code suggestions! Thank you